examples/dreambooth/train_dreambooth.py

```diff
@@ -0,0 +1,543 @@
+            padding="max_length",
+            truncation=True,
+            max_length=self.tokenizer.model_max_length,
+            return_tensors="pt",
+        ).input_ids[0]
+
+        if self.class_data_root:
+            class_image = Image.open(self.class_images_path[index % self.num_class_images])
+            if not class_image.mode == "RGB":
+                class_image = class_image.convert("RGB")
+            example["class_images"] = self.transform_image(class_image)
+            example["class_prompt_ids"] = self.tokenizer(
+                self.class_prompt,
+                padding="max_length",
+                truncation=True,
+                max_length=self.tokenizer.model_max_length,
+                return_tensors="pt",
+            ).input_ids[0]
+
+        return example
+
+
+def get_full_repo_name(model_id: str, organization: Optional[str] = None, token: Optional[str] = None):
+    if token is None:
+        token = HfFolder.get_token()
+    if organization is None:
+        username = whoami(token)["name"]
+        return f"{username}/{model_id}"
+    else:
+        return f"{organization}/{model_id}"
+
+
+def main():
+    args = parse_args()
+    logging_dir = Path(args.output_dir, args.logging_dir)
+
+    accelerator = Accelerator(
+        gradient_accumulation_steps=args.gradient_accumulation_steps,
+        mixed_precision=args.mixed_precision,
+        log_with="tensorboard",
+        logging_dir=logging_dir,
+    )
+
+    if args.seed is not None:
+        set_seed(args.seed)
+
+    if not args.without_prior_preservation:
+        class_images_dir = Path(args.class_data_dir)
+        if not class_images_dir.exists():
+            class_images_dir.mkdir(parents=True)
+        cur_class_images = len(list(class_images_dir.iterdir()))
+
+        if cur_class_images < args.num_class_images:
+            sd_model = StableDiffusionPipeline.from_pretrained(
+                args.pretrained_model_name_or_path, use_auth_token=args.use_auth_token
+            )
+            sd_model = accelerator.prepare(sd_model)
+            sd_model.to(accelerator.device)
+
+            num_new_images = args.num_class_images - cur_class_images
+            logger.info(f"Number of class images to sample: {num_new_images}.")
+            total_prompts = [args.class_prompt] * num_new_images
+            batch_prompts = [
+                total_prompts[x : x + args.sample_batch_size] for x in range(0, num_new_images, args.sample_batch_size)
+            ]
+
+            img_id = cur_class_images
+            for text in tqdm(
+                batch_prompts, desc="Generating class images", disable=not accelerator.is_local_main_process
+            ):
+                with torch.no_grad():
+                    images = sd_model(text, height=512, width=512, num_inference_steps=50)["sample"]
+
+                for image in images:
+                    image.save(class_images_dir / f"{img_id}.jpg")
+                    img_id += 1
```

Comment:
in multi-gpu setting this will run all gpu. We can only run this on one GPU or handle multi-gpu here. 
Let me know if you need help here.